grug/app/src/abci.rs

```diff
@@ -82,19 +81,7 @@ where
         let max_tx_bytes = req.max_tx_bytes.try_into().unwrap_or(0);
         let txs = self
             .do_prepare_proposal(req.txs.clone(), max_tx_bytes)
-            .unwrap_or_else(|err| {
-                // For the sake of liveness, in case proposal preparation fails,
-                // we fall back to the naive strategy instead of panicking.
-                #[cfg(feature = "tracing")]
-                error!(
-                    err = err.to_string(),
-                    "Failed to prepare proposal! Falling back to naive preparer."
-                );
-
-                NaiveProposalPreparer
-                    .prepare_proposal(QuerierWrapper::new(&NaiveQuerier), req.txs, max_tx_bytes)
-                    .unwrap()
-            });
+            .unwrap();
```

Comment:
If you don't expect `do_prepare_proposal` to ever fail, then it shouldn't return a `Result`, and we shouldn't need to unwrap here.

Can you rewrite the `do_prepare_proposal` function in this way:

```rust
pub fn do_prepare_propsoal() -> Vec<Bytes> {
    _do_prepare_proposal().unwrap_or_else(|| {
        // fallback to naive preparer
    })
}

fn _do_prepare_propsoal() -> AppResult<Vec<Bytes>> {
    // things that may fail
}
```